test_rclcpp/test/test_avoid_ros_namespace_conventions_qos.cpp

```diff
@@ -44,11 +44,11 @@ TEST(
   int counter = 0;
   auto callback =
     [&counter](test_rclcpp::msg::UInt32::ConstSharedPtr msg,
-      const rmw_message_info_t & info) -> void
+      const rclcpp::MessageInfo & info) -> void
```

Comment:
Is the previous callback signature still available?

This change is fine, but I'm just wondering if users that update to Foxy will see a warning being log or a build failure.

Reply:
No. It is a build failure. I'm going to add a note to the migration guide for Foxy.

Reply:
Ok, I'm not sure if deprecation is possible or not, but if not adding a note to the migration notes sounds fine.

Reply:
See: https://github.com/ros2/ros2_documentation/pull/606

Reply:
I can work on trying to support the old signature, but I thought it was a) uncommonly used and b) made the signature resolution a bit trickier since it changes the type but now produces multiple signatures with the same arity.

Reply:
> a) uncommonly used

Yeah, that's true.

b) made the signature resolution a bit trickier since it changes the type but now produces multiple signatures with the same arity.

Ok, sounds good.